main.go

```diff
@@ -182,23 +160,67 @@ func (s *slackOutput) ProcessMessage(rawmsg []byte) (msg []byte, tags []string,
 
 // SendBatch is called once per batch per tag and delivers the message to slack
 func (s *slackOutput) SendBatch(batch [][]byte, tag string) error {
-	failedMessages := [][]byte{}
-	var lastError error
+	// Decode the tag
+	msgTag := slackTag{}
+	err := json.Unmarshal([]byte(tag), &msgTag)
+	if err != nil {
+		return fmt.Errorf("Failed to decode message tag '%s': %s", tag, err.Error())
+	}
+
+	// Messages are batched by channel,user, & icon so just send one slack message
+	// per batch with each message on a new line
+	longMessages := [][]byte{}
+	texts := []string{}
+	length := 0
+	tmpMessages := []string{}
+
+	// limit total message length by size
+	for _, b := range batch {
+		msg := string(b)
+
+		// No single message can be longer than the max (this is a `>=` to account for the newline)
+		if len(msg) >= MaxMessageLength {
+			longMessages = append(longMessages, b)
+			continue
+		}
+
+		if length+len(msg)+1 > MaxMessageLength {
+			texts = append(texts, strings.Join(tmpMessages, "\n"))
+			tmpMessages = []string{}
+			length = 0
+		}
+		tmpMessages = append(tmpMessages, msg)
+		length += len(msg) + 1
+	}
 
-	for _, msg := range batch {
-		err := s.sendMessage(msg, tag)
+	// Make sure the final batch gets added
+	if len(tmpMessages) > 0 {
+		texts = append(texts, strings.Join(tmpMessages, "\n"))
```

Comment:
Why not concat `texts` into one long string since they're all going to the same channel/icon/user-name?